project_restorer/src/main/scala/PlutoCoreMessageProcessor.scala

```diff
@@ -36,28 +52,33 @@ class PlutoCoreMessageProcessor(mxsConfig:MatrixStoreConfig)(implicit mat:Materi
       .run()
   }
 
-
-  def searchAssociatedMedia(projectId: Int, vault: Vault)  = {
-   for {
-     result <- filesByProject(vault, projectId.toString)
-   } yield result
+  def getNearLineResults(projectId: Int) = {
+    matrixStoreBuilder.withVaultFuture(mxsConfig.nearlineVaultId) { vault =>
+      searchAssociatedNearlineMedia(projectId, vault).map(Right.apply)
+    }
   }
 
+  def getOnlineResults(projectId: Int) = {
+    searchAssociatedOnlineMedia(projectId, vidispineCommunicator).map(Right.apply)
+  }
 
-  def handleStatusMessage(updateMessage: ProjectUpdateMessage, routingKey: String, framework: MessageProcessingFramework):Future[Either[String, Json]] = {
-       matrixStoreBuilder.withVaultFuture(mxsConfig.nearlineVaultId) {vault =>
-        searchAssociatedMedia(updateMessage.id, vault).map(results=> {
-
-          if(results.length < 10000 ){
-            framework.bulkSendMessages(routingKey, results)
-            val msg = RestorerSummaryMessage(updateMessage.id, ZonedDateTime.now(), updateMessage.status, results.length, 0)
-            Right(msg.asJson)
-          }
-          else {
-            throw new RuntimeException("Too many files attached to project")
+  def handleStatusMessage(updateMessage: ProjectUpdateMessage, routingKey: String, framework: MessageProcessingFramework): Future[Either[String, MessageProcessorReturnValue]] = {
+    Future.sequence(Seq(getNearLineResults(updateMessage.id), getOnlineResults(updateMessage.id))).map(allResults => {
+      (allResults.head, allResults(1)) match {
+        case (Right(nearlineResults), Right(onlineResults)) =>
+          if (nearlineResults.length < 10000 && onlineResults.length < 10000) {
+            framework.bulkSendMessages(routingKey, nearlineResults)
+            framework.bulkSendMessages(routingKey, onlineResults)
+            val msg = RestorerSummaryMessage(updateMessage.id, ZonedDateTime.now(), updateMessage.status, numberOfAssociatedFilesNearline = nearlineResults.length, numberOfAssociatedFilesOnline = onlineResults.length)
+            Right(MessageProcessorConverters.contentToMPRV(msg.asJson))
```

Comment:
I thought that this should work as an implicit but maybe i am wrong

Reply:
I thought so to, but I couldn't get it to work

Reply:
fair enough